nova-benches/benches/shared/mod.rs

```diff
@@ -4,6 +4,7 @@ use ark_relations::r1cs::{ConstraintSystemRef, SynthesisError};
 use nexus_nova::StepCircuit;
 use std::marker::PhantomData;
 
+#[allow(dead_code)]
```

Comment:
This shouldn't be dead code.

Reply:
Since this is a shared code, it is used in other benches, however I don't use the `NUM_WARMUP_STEPS` in my code, so I turn off the clippy warning in compilation.

Reply:
True I guess it is dead for that individual build... I mean it's all test code so I guess that's fine...

Reply:
coincidently, the value of NUM_WARMUP_STEPS is 10, the minimum I need in criterion sample size is 10, I can plug in this value in my code and get rid of the `allow[dead_code]`, but the variable name doesn't quite describe what the situation.